src/Strategy/Maven/DepTree.hs

```diff
@@ -47,27 +42,100 @@ import Text.Megaparsec.Char (space1)
+      -- failure, since we can still get partial results.
+      --
+      -- TODO: We should emit a warning on non-zero exit.
+      cmdAllowErr = Always
+    }
 
-analyze :: (Has Exec sig m, Has ReadFS sig m, Has Diagnostics sig m, Has (Lift IO) sig m) => Path Abs Dir -> m (Graphing Dependency, GraphBreadth)
+analyze ::
+  ( Has Exec sig m
+  , Has ReadFS sig m
+  , Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  ) =>
+  Path Abs Dir ->
+  m (Graphing Dependency, GraphBreadth)
 analyze dir = do
-  -- We generate files with a random prefix so we don't pick up files from previous runs.
-  -- If we did pick them up, they're likely not enough of a conflict to worry about, but
-  -- it's better than not doing it.
-  randIdent <- abs <$> sendIO randomIO
-  _ <- context "Running maven 'dependency:tree' plugin" $ execThrow dir $ deptreeCmd randIdent
-  graphFiles <- context "Locating maven output files" $ findDepTreeOutputs dir randIdent
-  graphs <- context "Parsing output files" $ traverse (readContentsParser parseDotGraph) graphFiles
-  -- TODO: Will this work correctly in the face of exceptions?
-  sendIO $ traverse_ removeFile graphFiles
+  -- Construct the Maven command invocation.
+  --
+  -- First, we need to determine whether there exists a `settings.xml` that we
+  -- need to use. We only look for this in the root (i.e. next to the `pom.xml`)
+  -- that constructed this analysis target.
+  --
+  -- TODO: For maximum magic, find all `settings.xml`s in the CWD, and try with
+  -- all of them until one works?
+  settingsFile <- mustRelFile "settings.xml"
+  let settingsPath = dir </> settingsFile
+  settingsExists <- doesFileExist settingsPath
+
+  -- Second, we need a temporary filepath to write results to.
+  tmpDir <- sendIO getTempDir
+  rand :: Int <- abs <$> sendIO randomIO
+  tmpFileName <- mustRelFile $ "fossa-deptree-" <> show rand <> ".dot"
+  let tmp = tmpDir </> tmpFileName
+
+  -- Then we run the command, parse the temporary file's output, and clean up
+  -- the temporary file.
+  graphs <-
+    execAndParse (if settingsExists then Just settingsPath else Nothing) tmp
+      `finally` sendIO (removeFile tmp)
   pure (buildGraph graphs, Complete)
-
-findDepTreeOutputs :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> Int -> m [Path Abs File]
-findDepTreeOutputs dir ident = flip walk' dir $ \_ _ files -> do
-  case findFileNamed (show ident <> outputFileName) files of
-    Nothing -> pure ([], WalkContinue)
-    Just file -> pure ([file], WalkContinue)
+  where
+    execAndParse ::
+      (Has Exec sig m, Has ReadFS sig m, Has Diagnostics sig m) =>
+      Maybe (Path Abs File) ->
+      Path Abs File ->
+      m [DotGraph]
+    execAndParse settingsFile tmpFile = do
+      _ <- context "Running 'mvn dependency:tree'" $ exec dir $ deptreeCmd settingsFile tmpFile
+      context "Parsing 'mvn dependency:tree' output" $ readContentsParser parseDotGraphs tmpFile
```

Comment:
[optional]: Putting this function inside a `where` block seems weird.  I'd prefer to see it at the top level, but that's just preference.

Reply:
This is intentional to make the `finally` apply to the whole action. I've added a comment explaining this.